client/components/editor/teaching-elements/Assessment/TextSelection/Selector.vue

```diff
@@ -29,44 +26,42 @@ export default {
   },
   data() {
     const selections = new Selections(this.correct);
-    const processNode = selections.processNode.bind(selections);
-    const mapper = curry(nodeMapper)(processNode);
     return {
       selections,
-      structure: generateStructure(this.content, mapper)
+      structure: generateStructure(this.content, (node, index) => {
+        const processNode = selections.processNode.bind(selections);
```

Comment:
You are aware that calling `.bind` every time creates new function aka you're gonna create `content.length` number of exactly the same functions 🙃